package.json

```diff
@@ -1,66 +0,0 @@
-{
```

Comment:
Talking to Pedro, we thought we'd probably need to keep the `package.json` file. For a few reasons:
- (1) would fix the problem of DEVELOPERS not being able to bump the version; if we pulled the version information from the `package.json` file... they could easily do that, with an `npm update`; they wouldn't need to publish the blade file;
- (2) we'd receive notifications from Dependabot, when a package is vulnerable;
- (3) the way a package is turned from NAME to URL would then become configurable;
- (4) people who want to use NPM would be able to use it, because they can run it and publish the assets that way;

For example:
- in `package.json` you'd find `"ckeditor4": "^4.20.1",` and in `package-lock.json` you'll find the actual installed version for that package;
- in `config\bassets.php` you'd have a line which turns that into a URL; the final URL (`https://cdnjs.cloudflare.com/ajax/libs/ckeditor/4.20.1/ckeditor.js`) would be constructed using the version in `package-lock.json`:

```php
<?php

return [
    // ...
        'basset_urls' => [ // other names: named_urls?! named_assets?!
            'ckeditor4' => 'https://cdnjs.cloudflare.com/ajax/libs/ckeditor/{version}/ckeditor.js'
        ],
    // ...
];
```


// can't we use dyno?!

Reply:
When you do `php artisan bassets:internalize` it will:
- arrive at `@basset('ckeditor4')`
- which looks in the config file and sees there's a URL for it
- in that URL, it turns `{version}` into the version in the `package-lock.json` file?

Then if the developer wants to override a version, they can:
- go to their `config/bassets.php` file, and change

```diff
        'basset_urls' => [ // other names: named_urls?! named_assets?!
-            'ckeditor4' => 'https://cdnjs.cloudflare.com/ajax/libs/ckeditor/{version}/ckeditor.js'
+            'ckeditor4' => 'https://cdnjs.cloudflare.com/ajax/libs/ckeditor/4.20.2/ckeditor.js'
        ],
```

Then Bassets will not find `{version}` in the URL, and it will just use the explicit version that was defined by the developer.

Reply:
```

    NOW WE HAVE:
    @basset('https://cdnjs.cloudflare.com/ajax/libs/ckeditor/4.20.1/ckeditor.js')

    HOW ABOUT INSTEAD OF:
    @basset('ckeditor4')                                => config/bassets/file_paths.php

    WE DO:
    @basset(config('backpack.assets.ckeditor4'))        => config/backpack/assets.php (or maybe js.php and css.php?!)
```

Reply:
We'll probably end up with:
```
    @basset(config('backpack.assets.crud'))        => config/backpack/assets.php (or maybe js.php and css.php?!)
    @basset(config('backpack.assets.pro'))        => config/backpack/assets.php (or maybe js.php and css.php?!)
    @basset(config('backpack.assets.filemanager'))        => config/backpack/assets.php (or maybe js.php and css.php?!)
    @basset(config('backpack.assets.devtools'))        => config/backpack/assets.php (or maybe js.php and css.php?!)
```

Reply:
Moving issue to https://github.com/Laravel-Backpack/basset/issues/8